validate/src/embed.rs

```diff
@@ -302,6 +302,100 @@ pub fn embed(embed: &Embed) -> Result<(), EmbedValidationError> {
     Ok(())
 }
 
+pub(crate) fn _embed(embed: &Embed) -> Result<usize, EmbedValidationError> {
```

Comment:
this duplication could be avoided with this, at the cost of overhead 
```suggestion
pub(crate) fn _embed(embed: &Embed) -> Result<usize, EmbedValidationError> {
    embed(embed)?;
    Ok(count(embed))
}

fn count(embed: &Embed) -> usize {
    let mut total = 0;

    if let Some(name) = embed.author.as_ref().map(|author| &author.name) {
        total += name.chars().count();
    }
    
    if let Some(description) = embed.description.as_ref() {
         total += description.chars().count();
    }
    // and so on for every field
    total
}
```